File: Maximum Area of Longest Diagonal Rectangle/Maximum Area of Longest Diagonal Rectangle.py
# Solution for Maximum Area of Longest Diagonal Rectangle in PY

class Solution:
    def areaOfMaxDiagonal(self, dimensions: List[List[int]]) -> int:
        maxDiag, maxArea = 0, 0
        n = len(dimensions)
        
        for l, w in dimensions:
            diag = l*l + w*w
            area = l * w
            if diag > maxDiag:
                maxDiag = diag
                maxArea = area
            
            elif diag == maxDiag:
                maxArea = max(maxArea, area)
            
        return maxArea

File: Maximum Area of Longest Diagonal Rectangle/Maximum Area of Longest Diagonal Rectangle.java
// Solution for Maximum Area of Longest Diagonal Rectangle in JAVA

class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        int maxDiag = 0, maxArea = 0;
        int n = dimensions.length;
        for (int i = 0; i < n; i++) {
            int l = dimensions[i][0];
            int w = dimensions[i][1];
            int diag = l*l + w*w;
            int area = l * w;
            if (diag > maxDiag) {
                maxDiag = diag;
                maxArea = area;
            } else if (diag == maxDiag) {
                maxArea = Math.max(maxArea, area);
            }
        }
        return maxArea;
    }
}
